lib/queue.js

```diff
@@ -139,116 +153,148 @@ Queue.prototype.takeNext = function(
   return this;
 };
 
-Queue.prototype.pause = function(
-  // Pause queue
-) {
+// Pause queue
+// This function is not completely implemented yet
+//
+// Returns: <this>
+Queue.prototype.pause = function() {
   this.paused = true;
-  // stub
   return this;
 };
 
-Queue.prototype.resume = function(
-  // Resume queue
-) {
+// Resume queue
+// This function is not completely implemented yet
+//
+// Returns: <this>
+Queue.prototype.resume = function() {
   this.paused = false;
-  // stub
   return this;
 };
 
-Queue.prototype.clear = function(
-  // Clear queue
-) {
+// Clear queue
+//
+// Returns: <this>
+Queue.prototype.clear = function() {
   this.count = 0;
   this.tasks = [];
   this.waiting = [];
   this.factors = {};
   return this;
 };
 
+// Set timeout interval and listener
+//   msec - <number>, process timeout for single item
+//   onTimeout - <Function>
+//
+// Returns: <this>
 Queue.prototype.timeout = function(
-  // Set timeout interval and listener
-  msec, // number, process timeout for single item
-  onTimeout = null // function, (item) => {}
+  msec,
+  onTimeout = null
 ) {
   this.processTimeout = msec;
   if (onTimeout) this.onTimeout = onTimeout;
   return this;
 };
 
-Queue.prototype.process = function(
-  // Set processing function
-  fn // function, processing (item, callback)
-) {
+// Set processing function
+//   fn - <Function>
+//     item - <Object>
+//     callback - <Function>
+//       err - <Error>
+//       result - <Object>
```

Comment:
I don't think this is limited to Objects only:
```suggestion
//       result - <any>
```